### DISCUSSION TITLE
Uncaught ReferenceError: Taro is not defined

### DISCUSSION THREAD

**Discussion by 845056166** [NervJS/taro#10833] [Q & A]
打包成h5后，路由跳转报错：
app.979e9.js:formatted:5733 Uncaught ReferenceError: Taro is not defined

我的navTo方法如下
`
import { navigateTo } from "@tarojs/taro";
export const navTo = (url: string, linkText?: string): void => {
  let fixurl = url;
  const isSecret = url.startsWith("/pages");
  const isHTTP = url.startsWith("http");
  if (isSecret) {
    fixurl = `${window.location.origin}${BASE_URL}${url}`;
  } else if (!isHTTP) {
    fixurl = `${window.location.origin}${url}`;
  }
if (isSecret) {
    navigateTo({ url });
  } else {
    window.location.href = fixurl;
  }
};`

不知道大家有没有遇到过

**Comment by luckyadam** [comment]
CLI 和项目依赖都升级到最新版本试试

**Reply by 845056166** [reply]
我不用解构的方式，把整个Taro都导进来，然后Taro. navigateTo就好了，感觉打包的时候有问题
